testsupport/tiers/checks.go

```diff
@@ -48,333 +47,234 @@ var (
-	verifyNsTypes(hostAwait.T, a.tierName, templateRefs, "dev")
-	return templateRefs
-}
-
-func (a *base1nsTierChecks) GetClusterObjectChecks() []clusterObjectsCheck {
-	return clusterObjectsChecks(
-		clusterResourceQuotaCompute(cpuLimit, "1750m", "7Gi", "15Gi"),
-		clusterResourceQuotaDeployments(),
-		clusterResourceQuotaReplicas(),
-		clusterResourceQuotaRoutes(),
-		clusterResourceQuotaJobs(),
-		clusterResourceQuotaServices(),
-		clusterResourceQuotaBuildConfig(),
-		clusterResourceQuotaSecrets(),
-		clusterResourceQuotaConfigMap(),
-		numberOfClusterResourceQuotas(),
-		idlers(43200, "dev"))
-}
-
-type baselargeTierChecks struct {
-	baseTierChecks
-}
-
-func (a *baselargeTierChecks) GetClusterObjectChecks() []clusterObjectsCheck {
-	return clusterObjectsChecks(
-		clusterResourceQuotaCompute(cpuLimit, "1750m", "16Gi", "15Gi"),
-		clusterResourceQuotaDeployments(),
-		clusterResourceQuotaReplicas(),
-		clusterResourceQuotaRoutes(),
-		clusterResourceQuotaJobs(),
-		clusterResourceQuotaServices(),
-		clusterResourceQuotaBuildConfig(),
-		clusterResourceQuotaSecrets(),
-		clusterResourceQuotaConfigMap(),
-		numberOfClusterResourceQuotas(),
-		idlers(43200, "dev", "stage"))
-}
-
-func (a *baselargeTierChecks) GetTierObjectChecks() []tierObjectCheck {
-	return []tierObjectCheck{nsTemplateTier(a.tierName, 90)}
-}
-
-type baseextendedTierChecks struct {
-	baseTierChecks
-}
-
-func (a *baseextendedTierChecks) GetTierObjectChecks() []tierObjectCheck {
-	return []tierObjectCheck{nsTemplateTier(a.tierName, 180)}
-}
-
-type baseextendedidlingTierChecks struct {
-	baseTierChecks
-}
-
-func (a *baseextendedidlingTierChecks) GetClusterObjectChecks() []clusterObjectsCheck {
-	return clusterObjectsChecks(
-		clusterResourceQuotaCompute(cpuLimit, "1750m", "7Gi", "15Gi"),
-		clusterResourceQuotaDeployments(),
-		clusterResourceQuotaReplicas(),
-		clusterResourceQuotaRoutes(),
-		clusterResourceQuotaJobs(),
-		clusterResourceQuotaServices(),
-		clusterResourceQuotaBuildConfig(),
-		clusterResourceQuotaSecrets(),
-		clusterResourceQuotaConfigMap(),
-		numberOfClusterResourceQuotas(),
-		idlers(518400, "dev", "stage"))
 }
 
-type basedeactivationdisabledTierChecks struct {
-	baseTierChecks
-}
-
-func (a *basedeactivationdisabledTierChecks) GetTierObjectChecks() []tierObjectCheck {
-	return []tierObjectCheck{nsTemplateTier(a.tierName, 0)}
+func getClusterObjectChecks(tierName string) func() []clusterObjectsCheck {
+	return func() []clusterObjectsCheck {
+		switch tierName {
+		case appstudio:
+			return clusterObjectsChecks(
+				clusterResourceQuotaCompute(cpuLimit, "1750m", "7Gi", "15Gi"),
+				clusterResourceQuotaDeployments(),
+				clusterResourceQuotaReplicas(),
+				clusterResourceQuotaRoutes(),
+				clusterResourceQuotaJobs(),
+				clusterResourceQuotaServices(),
+				clusterResourceQuotaBuildConfig(),
+				clusterResourceQuotaSecrets(),
+				clusterResourceQuotaConfigMap(),
+				numberOfClusterResourceQuotas(),
+				idlers(43200, ""))
```

Comment:
I think we can reduce even more duplication by including the common checks in the `clusterObjectsChecks` func and only provide the differentiating checks here
```suggestion
			return clusterObjectsChecks(
				clusterResourceQuotaCompute(cpuLimit, "1750m", "7Gi", "15Gi"),
				idlers(43200, ""))
```
```suggestion
			return clusterObjectsChecks(
				clusterResourceQuotaCompute(cpuLimit, "1750m", "7Gi", "15Gi"),
				clusterResourceQuotaDeployments(),
				clusterResourceQuotaReplicas(),
				clusterResourceQuotaRoutes(),
				clusterResourceQuotaJobs(),
				clusterResourceQuotaServices(),
				clusterResourceQuotaBuildConfig(),
				clusterResourceQuotaSecrets(),
				clusterResourceQuotaConfigMap(),
				numberOfClusterResourceQuotas(),
				idlers(43200, ""))
```

Reply:
good point! done in a013cb3